exp/query/api.go

```diff
@@ -112,6 +116,12 @@ func (a *API) Search(ctx context.Context, req *SearchRequest) (*SearchResponse,
 		return nil, fmt.Errorf("tenant %q is not offloaded, %w", req.Tenant, ErrInvalidTenant)
 	}
 
+	// TODO come up with a better way to ensure this doesn't run until the gRPC serving is active
```

Comment:
If you have any thoughts for how to solve this, let me know, this `sync.Once`/`start` stuff is obviously not how this should be